### DISCUSSION TITLE
Is dynamic (runtime) creation and destruction of reactors possible?

### DISCUSSION THREAD

**Discussion by godotdot** [lf-lang/lingua-franca#1931] [Q&A] [answered]
Does lingua-franca provide the ability to dynamically create (and destroy) reactors at runtime?

Two example use-cases might be:
1 - In a program allowing plugins, assign each plugin to its own reactor.
2 - In the same scenario, allow plugin code to create reactors.

If so, is there any guidance on how might one go about doing this?
If not, are there any plans to add this capability?

**Comment by lhstrh** [comment, marked as answer]
Currently, these kinds of runtime mutations are generally not supported, but we are experimenting with them in the TypeScript target. We soon hope to have a prototype that we have enough confidence in to where we can furnish other targets with this capability as well.

**Comment by godotdot** [comment]
Hey, thank you for the quick answer.  That sounds great, I'll look forward to it.

**Reply by lhstrh** [reply]
I'll post an update in this thread once we have some new developments on this front. I expect to have some to share pretty soon :rocket:

**Reply by godotdot** [reply]
That would be awesome; I wanted to ask for that but I didn't want to be a pain, so thank you for this.
